[Bug] `nvidia/Gemma-4-26B-A4B-NVFP4` fails `assert M % 128 == 0` on FlashInfer TRT-LLM NVFP4 MoE at `-tp 4` (Blackwell/B200)

> **Scope:** verified for `nvidia/Gemma-4-26B-A4B-NVFP4` on B200 (SM100), `flashinfer_trtllm`
> backend. This is **NOT "TRT-LLM fails at all TP>1"** — it is **model-dimension-specific**. It
> depends entirely on whether this model's gate+up row dim is already 128-aligned at the chosen TP
> (table below). Other models / TPs whose dim is already 128-aligned run fine; this is not a blanket
> TP>1 failure.

## What happens
On **B200 (SM100)**, serving `nvidia/Gemma-4-26B-A4B-NVFP4` at `--tensor-parallel-size 4` fails
during `process_weights_after_loading` (engine core never starts) with:

```
File ".../flashinfer/utils.py", line 882, in get_shuffle_matrix_sf_a_row_indices
    M, K = input_tensor.shape
    assert M % 128 == 0
AssertionError
```

Reproduced with **`nvidia/Gemma-4-26B-A4B-NVFP4` at `--tensor-parallel-size 4`** (tp1 works).

## Repro (serve sanity, minimal — B200)
```bash
vllm serve nvidia/Gemma-4-26B-A4B-NVFP4 \
  --tensor-parallel-size 4 \
  --quantization modelopt \
  --trust-remote-code
# -> AssertionError: assert M % 128 == 0   (model never serves)

# WORKS: --tensor-parallel-size 1   (nothing sharded; intermediate stays 128-aligned)
```

## Hardware scope
**Blackwell only (SM100 / B200).** The assert is reached only via the `flashinfer_trtllm`
NVFP4 MoE backend, which is first in `select_nvfp4_moe_backend`'s `AVAILABLE_BACKENDS` and is
**Blackwell-only**. On SM80/SM89/SM90 (A100/H100) TRT-LLM is unavailable, so the MoE runs on
`vllm_cutlass`/`marlin` and this assert never fires.

## Full call path (where the assert comes from)
```
modelopt.py:process_weights_after_loading
 -> fused_moe/oracle/nvfp4.py:convert_to_nvfp4_moe_kernel_format
 -> quantization/utils/flashinfer_fp4_moe.py:prepare_nvfp4_moe_layer_for_fi_or_cutlass
 -> quantization/utils/flashinfer_fp4_moe.py:prepare_static_weights_for_trtllm_fp4_moe
 -> flashinfer/fused_moe/core.py:_maybe_get_cached_w3_w1_permute_indices
 -> flashinfer/utils.py:get_shuffle_matrix_sf_a_row_indices  ->  assert M % 128 == 0
```
`get_shuffle_matrix_sf_a_row_indices` requires the **row dimension `M` of the w13 (gate+up)
scale-factor matrix** to be a multiple of 128 (`epilogue_tile_m = 128`). It pads `K` internally
but **not** `M` — the caller must pre-align `M`.

## Root cause (and where the `2x` comes from)
`align_fp4_moe_weights_for_fi` (quantization/utils/flashinfer_utils.py) pads the intermediate:
```python
padded_intermediate = round_up(intermediate, min_alignment)   # min_alignment=16 when gated
up_mult = 2 if is_act_and_mul else 1                          # 2 for gated (SwiGLU/GeGLU)
padded_gate_up_dim = up_mult * padded_intermediate            # == w13 scale-matrix M
```
The **`2x` is  ostensibly the gate+up fusion** of a gated FFN. 

From this checkpoint's weights: `gate_proj.weight = [704, 1408]` and `up_proj.weight = [704, 1408]` are two separate projections,
each with output dim `704 = moe_intermediate_size` (1408 = `hidden_size/2`, FP4-packed). vLLM's
FusedMoE **stacks gate+up into `w13`** → output dim `704 + 704 = 1408 = 2 × intermediate`; their
scale tensors `gate_proj.weight_scale = [704, 176]` likewise fuse to `[1408, 176]`. At tp4 the
expert intermediate (704) shards to 176/rank, so the fused `w13` scale has `2 × 176 = 352` rows —
that is exactly the `M` the FlashInfer shuffle asserts on (`352 % 128 = 96` → fail). The vLLM code
encodes this fusion: `gemm1_intermediate_size = 2 * intermediate_size if is_gated_activation`.

The caller passes `min_alignment = 16 if is_gated else 128` (`flashinfer_fp4_moe.py`). The `16`
is the **NVFP4 scale-block size** (`num_elts_per_sf=16` — the FP4 quantization-scale granularity,
see the `//16` reshapes), introduced with the TRT-LLM MoE support (#33506). It aligns the quant
scale blocks but is **a different, weaker constraint than the shuffle's 128-row epilogue tile** —
so for gated, `M = 2 * round_up(intermediate, 16)` is guaranteed only to be a multiple of **32**.
**This does not mean TRT-LLM fails at all TP>1 — it is purely a function of the model's
dimensions.** The assert fires *iff* `round_up(rank_local_intermediate, 16)` is not a multiple of
64; for the vast majority of models/TPs it already is, and they run correctly. Gemma-4-26B-A4B
(`moe_intermediate_size = 704`) happens to violate it at tp2/4/8 but **not tp1** (see table). The
non-gated path already uses `min_alignment=128` and is fine. Not covered by the Marlin padding
consolidation (#45295), which is Marlin-only.

## Proposed fix
Use `min_alignment = 64` for gated activations so `2 * padded_intermediate` is a multiple of 128.
Padded rows are zero in both weights and scales, so outputs are unchanged.

## Environment
- vLLM `main`; FlashInfer (bundled)
- GPU: NVIDIA B200 (SM100); `--quantization modelopt`, NVFP4 MoE, gated activation

--- a/vllm/model_executor/layers/quantization/utils/flashinfer_fp4_moe.py
+++ b/vllm/model_executor/layers/quantization/utils/flashinfer_fp4_moe.py
@@ -337,7 +337,13 @@ def prepare_nvfp4_moe_layer_for_fi_or_cutlass(
         layer.moe_config.hidden_dim = padded_hidden
 
         # Align weights for FI NVFP4 MoE kernels.
-        min_alignment = 16 if is_gated else 128
+        # FlashInfer's TRT-LLM block-scale shuffle asserts the gate/up row dim
+        # (= up_mult * padded_intermediate, up_mult=2 when gated) is a multiple of
+        # 128. So gated needs padded_intermediate % 64 (2*64=128); the old value 16
+        # left 2*intermediate a multiple of only 32, so an NVFP4 MoE whose rank-local
+        # intermediate is not 128-aligned at TP>1 (e.g. Gemma-4-26B-A4B at tp4) hit
+        # `assert M % 128 == 0`. Padded rows are zero -> outputs unchanged.
+        min_alignment = 64 if is_gated else 128
         w13, w13_scale, w2, w2_scale, padded_intermediate = (
             align_fp4_moe_weights_for_fi(
                 w13, w13_scale, w2, w2_scale, is_act_and_mul, min_alignment

